audio/midi/fluid/sfont3.cpp

```diff
@@ -10,40 +10,45 @@ namespace FluidS {
 //   decompressOggVorbis
 //---------------------------------------------------------
 
-bool Sample::decompressOggVorbis(char* src, int size)
+bool Sample::decompressOggVorbis(char* src, unsigned int size)
       {
       AudioFile af;
       QByteArray ba(src, size);
 
       start = 0;
       end   = 0;
       if (!af.open(ba)) {
-            qDebug("Sample::decompressOggVorbis: open failed: %s", af.error());
+            qDebug("SoundFont(%s) Sample(%s) decompressOggVorbis: open failed: %s", qPrintable(sf->get_name()), name, af.error());
             return false;
             }
-      int frames = af.frames();
+      unsigned int frames = af.frames();
       data = new short[frames * af.channels()];
-      if (frames != af.readData(data, frames)) {
-            qDebug("Sample read failed: %s", af.error());
+      if (frames != (unsigned int)af.readData(data, frames)) {
+            qDebug("SoundFont(%s) Sample(%s) read failed: %s", qPrintable(sf->get_name()), name, af.error());
             delete[] data;
             data = 0;
+            return false;
             }
-      end = frames - 1;
+      // cf. https://musescore.org/en/node/89216#comment-1068379 and following
+      end = frames;
 
-      if (loopend > end ||loopstart >= loopend || loopstart <= start) {
+      // loop is fowled?? (cluck cluck :)
+      if (loopend > end || loopstart >= loopend || loopstart <= start) {
+            qWarning("SoundFont(%s) Sample(%s) start(%u) startloop(%u) endloop(%u) end(%u) fowled (broken soundfont?), fixing up",
+                     qPrintable(sf->get_name()), name, start, loopstart, loopend, end);
             /* can pad loop by 8 samples and ensure at least 4 for loop (2*8+4) */
             if ((end - start) >= 20) {
                   loopstart = start + 8;
-                  loopend = end - 8;
+                  loopend   = end - 8;
                   }
-            else { // loop is fowled, sample is tiny (can't pad 8 samples)
+            else {      // loop is fowled, sample is tiny (can't pad 8 samples)
```

Comment:
better drop these and the following whitespace changes, they probably won't work for master's style anyhow

Reply:
These are here because they are *precise* copy/paste from `sfont.cpp` replacing what was in `sfont3.cpp` before; I’ll handle master similarily. But here, I want to make them identical.

Reply:
Ah, I see.